cpp/src/dbscan/vertexdeg/algo.h

```diff
@@ -60,14 +60,21 @@ void launcher(const ML::cumlHandle_impl& handle, Pack<value_t, index_t> data,
   auto fused_op = [vd, n] __device__(index_t global_c_idx, bool in_neigh) {
     // fused construction of vertex degree
     index_t batch_vertex = global_c_idx - (n * (global_c_idx / n));
-    atomicAdd(vd + batch_vertex, in_neigh);
-    atomicAdd(vd + n, in_neigh);
+
+    if (sizeof(index_t) == 4) {
+      atomicAdd((int *)(vd + batch_vertex), (int)in_neigh);
```

Comment:
wow! painful that this needs to be split!!
seems weird that one branch is unsigned and the other is not... why not int and long?